code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -465,3 +482,244 @@
+	SEND_SOUND(S, 'sound/magic/demon_dies.ogg')
+	message_admins("[ADMIN_LOOKUPFLW(S)] has been made into a slaughter demon by dynamic.")
+	log_game("[key_name(S)] was spawned as a slaughter demon by dynamic.")
+	return S
+
+//////////////////////////////////////////////
+//                                          //
+//               ABDUCTORS                  //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/abductors
+	name = "Abductors"
+	config_tag = "abductors"
+	antag_flag = ROLE_ABDUCTOR
+	// Has two antagonist flags, in fact
+	enemy_roles = list("AI", "Cyborg", "Security Officer", "Warden","Detective","Head of Security", "Captain")
+	required_enemies = list(3,3,2,2,1,1,0,0,0,0)
+	required_candidates = 2
+	weight = 8
+	cost = 10
+	requirements = list(80,80,70,50,40,30,30,20,15,15)
+	high_population_requirement = 15
+	var/datum/team/abductor_team/team
+	repeatable_weight_decrease = 4
+	repeatable = TRUE
+
+/datum/dynamic_ruleset/midround/from_ghosts/abductors/acceptable(population=0, threat=0)
+	if (locate(/datum/dynamic_ruleset/roundstart/nuclear) in mode.executed_rules)
+		return FALSE // Unavailable if nuke ops were already sent at roundstart. yes, this is intentional for abductors too.
+	return ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/abductors/ready(forced = FALSE)
+	if(required_candidates > (dead_players.len + list_observers.len))
+		return FALSE
+	team = new /datum/team/abductor_team
+	if(team.team_number > ABDUCTOR_MAX_TEAMS)
+		return FALSE
+	return ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/abductors/finish_setup(mob/new_character, index)
+	switch(index)
+		if(1) // yeah this seems like a baffling anti-pattern but it's actually the best way to do this, shit you not
+			var/mob/living/carbon/human/agent = new_character
+			agent.mind.add_antag_datum(/datum/antagonist/abductor/agent, team)
+			log_game("[key_name(agent)] has been selected as [team.name] abductor agent.")
+		if(2)
+			var/mob/living/carbon/human/scientist = new_character
+			scientist.mind.add_antag_datum(/datum/antagonist/abductor/scientist, team)
+			log_game("[key_name(scientist)] has been selected as [team.name] abductor scientist.")
+
+//////////////////////////////////////////////
+//                                          //
+//              SPACE NINJA                 //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja
+	name = "Space Ninja"
+	config_tag = "ninja"
+	antag_flag = ROLE_NINJA
+	enemy_roles = list("Security Officer","Head of Security","Captain","AI","Cyborg")
+	required_enemies = list(3,2,2,2,2,1,1,1,1,0)
+	required_candidates = 1
+	weight = 4
+	cost = 15
+	requirements = list(101,101,101,90,80,70,60,50,40,30)
```

Comment:
why is ninja so expensive

Reply:
didn't get caught in my balance passes, probably, will tweak